plugin/metrics/prometheus/metricsstore/reader.go

```diff
@@ -264,8 +267,33 @@ func getHTTPRoundTripper(c *config.Configuration, logger *zap.Logger) (rt http.R
 		TLSHandshakeTimeout: 10 * time.Second,
 		TLSClientConfig:     ctlsConfig,
 	}
-	return bearertoken.RoundTripper{
-		Transport:       httpTransport,
-		OverrideFromCtx: true,
-	}, nil
+
+	token := ""
+	if c.TokenFilePath != "" {
+		if c.AllowTokenFromContext {
+			logger.Warn("Token file and token propagation are both enabled, token from file won't be used")
+		}
+		tokenFromFile, err := loadToken(c.TokenFilePath)
+		if err != nil {
+			return nil, err
+		}
+		token = tokenFromFile
+	}
+	if token != "" || c.AllowTokenFromContext {
+		transport = bearertoken.RoundTripper{
+			Transport:       httpTransport,
+			OverrideFromCtx: c.AllowTokenFromContext,
+			StaticToken:     token,
+		}
+	}
+
+	return transport, nil
```

Comment:
I think we can simplify this to:
```
return bearertoken.RoundTripper{
		Transport:       httpTransport,
		OverrideFromCtx: c.AllowTokenFromContext,
		StaticToken:     token,
	}, nil
```

Also, the above suggestion should fix a bug where it's possible for `transport` to not be set if `token == ""`, which means we lose our `httpTransport` configuration, including TLS configuration (it would be nice if there's a way to test TLS enablement).